homeassistant/components/switch/tplink.py

```diff
@@ -60,3 +65,27 @@ def turn_on(self, **kwargs):
     def turn_off(self):
         """Turn the switch off."""
         self.smartplug.state = 'OFF'
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        _LOGGER.debug("Updating TP-Link energy meter data")
+
+        emeter_readings = self.smartplug.get_emeter_realtime()
```

Comment:
You are not allowed to fetch data from an entity inside a property. Instead you should cache this data inside the `update` method.